File: DynamoDB-SDK-Examples/node.js/WorkingWithTables/enable_streams.js
// The StreamViewType can be any of the following values 'NEW_IMAGE'|'OLD_IMAGE'|'NEW_AND_OLD_IMAGES'|'KEYS_ONLY',
// but if you want to use Global Tables, it must be set to NEW_AND_OLD_IMAGES.

const {DynamoDBClient, UpdateTableCommand } = require('@aws-sdk/client-dynamodb');

const REGION = "us-west-2";
const TableName = "Music";

const dbclient = new DynamoDBClient({ region: REGION });

async function enableStreams() {
    const params = {
        TableName: TableName,
        StreamSpecification: {
            StreamEnabled: true,
            StreamViewType: "NEW_AND_OLD_IMAGES",
        },
    };
    return await dbclient.send( new UpdateTableCommand(params));
    // await waitForTableExists({ maxWaitTime: 20, maxDelay: 5, minDelay: 1 }, { TableName: TableName }  );
    // this code is commented out as there is something wrong with the waiters in v3. Once that is fixed, this code will be updated.
}

enableStreams()
    .then((data) => console.log(data))
    .catch((error) => console.log("An error occurred while enabling streams:" + ' ' + error.message ));

File: examples/SDK/golang/WorkingWithTables/enableStreams/enableStreams.go
package main

import (
    "github.com/aws/aws-sdk-go/aws"
    "github.com/aws/aws-sdk-go/aws/session"
    "github.com/aws/aws-sdk-go/service/dynamodb"
    "fmt"
)

var tableName = "Music"
var awsRegion = "us-west-2"

func getSession() (*session.Session) {
    sess := session.Must(session.NewSessionWithOptions(session.Options{
        SharedConfigState: session.SharedConfigEnable,
        // Provide SDK Config options, such as Region and Endpoint
        Config: aws.Config{
            Region: aws.String(awsRegion),
        },
    }))

    return sess
}

func enableStreams() error {
    dynamoDBClient := dynamodb.New(getSession())

    _, err := dynamoDBClient.UpdateTable(&dynamodb.UpdateTableInput{
        StreamSpecification: &dynamodb.StreamSpecification{
            StreamEnabled:  aws.Bool(true),
            //'NEW_IMAGE'|'OLD_IMAGE'|'NEW_AND_OLD_IMAGES'|'KEYS_ONLY'
            StreamViewType: aws.String("NEW_AND_OLD_IMAGES"),
        },
        TableName: aws.String(tableName),
    })

    if err != nil {
        fmt.Println("Error enabling streams", err)
        return err
    }

    return nil
}

func main() {
    fmt.Println("Enabling streams in table ...")
    enableStreams()
    fmt.Println("Finished ...")
}
